course_content/notebooks/brief_iris_intro.ipynb

```diff
@@ -0,0 +1,2214 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "pot_temperature_constraint = iris.Constraint('air_potential_temperature',\n",
+    "                                             model_level_number=10)\n",
+    "print cubes.extract(pot_temperature_constraint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can pass a list of possible values, and even combine two constraints with ``&``:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "print cubes.extract('air_potential_temperature' & \n",
+    "                    iris.Constraint(model_level_number=[4, 10]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can define arbitrary functions that operate on each cell of a coordinate. This is a common thing to do for floating point coordinates, where exact equality is non-trivial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def less_than_10(cell):\n",
+    "    \"\"\"Return True for values that are less than 10.\"\"\"\n",
+    "    return cell < 10\n",
+    "\n",
+    "print cubes.extract(iris.Constraint('air_potential_temperature',\n",
+    "                                    model_level_number=less_than_10))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Time Constraints\n",
+    "\n",
+    "It is common to want to build a constraint for time. This can be achieved by comparing cells containing datetimes. The functionality can be enabled globally within the session (and will be enabled by default in v2.0, the next release of Iris) with:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "iris.FUTURE.cell_datetime_objects = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are a few different approaches for producing time constraints in Iris. We will focus here on one approach for constraining on time in Iris. \n",
+    "\n",
+    "This approach allows us to access individual components of cell datetime objects and run comparisons on those:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "time_constraint = iris.Constraint(time=lambda c: c.point.hour == 11)\n",
+    "print air_pot_temp.extract(time_constraint).summary(True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise 2\n",
+    "\n",
+    "The following function tells us whether or not a cube has cell methods:\n",
```

Comment:
Or I could quickly introduce cell methods here and then move on with the exercise. The point of this bit of the exercise is using `cube_func`; cell methods just happen to be a reasonably accessible thing to construct a `cube_func` for...